assembly/README.md

```diff
@@ -1,42 +1,79 @@
-# Miden assembly
+# Miden Assembly
+
 This crate contains Miden assembler.
 
 The purpose of the assembler is to compile [Miden assembly](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html) source code into a Miden VM program (represented by `Program` struct). The program can then be executed on Miden VM [processor](../processor).
 
-## Compiling assembly code
-To compile Miden assembly source code into a program for Miden VM, you first need to instantiate the assembler, and then call its `compile()` method. This method takes the following arguments:
+## Compiling Miden Assembly
+
+To compile Miden assembly source code into a program for Miden VM, you first need to instantiate the assembler, and then call one of its provided compiler
+methods, e.g. `compile`.
 
-* `source: &str` - a reference to a string containing Miden assembly source code.
+The `compile` method takes the source code of an executable module as a string,
+and either compiles it to a `Program`, or returns an error if the program is
+invalid in some way. The error type returned can be pretty-printed to show
+rich diagnostics about the source code from which an error is derived, when
+applicable, much like the Rust compiler.
```

Comment:
It seems your editor is not picking `.editorconfig`:

https://github.com/0xPolygonMiden/miden-vm/blob/3a957f7c90176914bda2139f74bff9e5700d59ac/.editorconfig#L13-L14

Reply:
Yeah I mentioned that in response to Bobbin's earlier note about a similar thing. I've got it configured to use 100 cols now, and went back and updated a bunch of stuff, but I didn't update this file.